subprojects/language-jvm/src/main/java/org/gradle/api/internal/tasks/compile/daemon/AbstractDaemonCompiler.java

```diff
@@ -73,15 +79,27 @@ public String getDisplayName() {
         public DefaultWorkResult compile() {
             return new DefaultWorkResult(compiler.execute(spec).getDidWork(), null);
         }
+
+        public File getExecutionWorkingDir() {
+            return executionWorkingDir;
+        }
+
+        public File getIdleWorkingDir() {
+            return idleWorkingDir;
+        }
     }
 
     public static class CompilerDaemonServer implements WorkerProtocol<WorkerCompileSpec<?>> {
         @Override
         public DefaultWorkResult execute(WorkerCompileSpec<?> spec) {
+            ProcessEnvironment processEnvironment = NativeServices.getInstance().get(ProcessEnvironment.class);
```

Comment:
Yes, see https://github.com/gradle/gradle/compare/pm/workers/compilation-workers-spike#diff-0ba58c695ab35c11d651adb85c855323.  I consider this to be part of the work to dog-food the worker api in compile tasks.